solidity/contracts/converter/types/liquidity-pool-v1/LiquidityPoolV1Converter.sol

```diff
@@ -147,11 +146,27 @@ contract LiquidityPoolV1Converter is LiquidityPoolConverter {
         else
             safeTransfer(_targetToken, _beneficiary, amount);
 
+        uint256 poolTokenSupply = ISmartToken(address(anchor)).totalSupply();
+        uint256 sourceReserveBalance = reserveBalance(_sourceToken);
+        uint256 targetReserveBalance = reserveBalance(_targetToken);
+        uint32 sourceReserveWeight = reserves[_sourceToken].weight;
+        uint32 targetReserveWeight = reserves[_targetToken].weight;
+
         // dispatch the conversion event
         dispatchConversionEvent(_sourceToken, _targetToken, _trader, _amount, amount, fee);
 
-        // dispatch rate updates
-        dispatchRateEvents(_sourceToken, _targetToken);
+        // dispatch token rate update event
+        uint256 rateN = targetReserveBalance.mul(sourceReserveWeight);

```

Comment:
I think it's still worth having a pure utility function that does instead of doing the calculation (and potentially missing something like the order) in every place.

Reply:
Fixed.